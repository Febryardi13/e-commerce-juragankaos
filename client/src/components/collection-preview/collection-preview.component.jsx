import React from 'react';
import { withRouter } from 'react-router-dom'

import CollectionItem from '../collection-item/collection-item.component'

import {
    CollectionPreviewContainer,
    TittleContainer,
    PreviewContainer
} from './collection-preview.styles'

export const CollectionPreview = ({ title, history, match, items, routeName }) => (
    <CollectionPreviewContainer>
        <TittleContainer onClick={() => history.push(`${match.url}/${routeName}`)}>
            {title.toUpperCase()}
        </TittleContainer>
        <PreviewContainer>
            {
                items
                .filter((item, index) => index < 4)
                .map(item => (
                   <CollectionItem key={item.id} item={item}/>
                ))
            }
        </PreviewContainer>
    </CollectionPreviewContainer>
)

export default withRouter (CollectionPreview)