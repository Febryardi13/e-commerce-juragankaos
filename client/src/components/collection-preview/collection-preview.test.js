import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { CollectionPreview } from './collection-preview.component'
configure({ adapter: new Adapter })

describe('CollectionPreview component', () => {
    let wrapper;
    let mockMatch;
    let mockHistory;
    const mockRouteName = 'hats';

    beforeEach(() => {
        mockMatch = {
            path: '/shop'
        }

        mockHistory = {
            push: jest.fn()
        }

        const mockProps = {
            match: mockMatch,
            history: mockHistory,
            routeName: mockRouteName,
            title: 'hats',
            items: []
        }

        wrapper = shallow(<CollectionPreview {...mockProps} />)
    })

    it('should render CollectionPreview compoenent', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should call history.push with the right string when TitleContainer clicked', () => {
        wrapper.find('TittleContainer').simulate('click')

        expect(mockHistory.push).toHaveBeenCalledWith(
            `${mockMatch.url}/${mockRouteName}`
        )
    })
})