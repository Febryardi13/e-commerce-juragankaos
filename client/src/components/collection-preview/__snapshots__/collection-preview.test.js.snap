// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render CollectionPreview compoenent 1`] = `ShallowWrapper {}`;
